tests/shared/src/test/scala-2/zio/schema/MetaSchemaSpec.scala

```diff
@@ -253,6 +268,27 @@ object MetaSchemaSpec extends ZIOSpecDefault {
           assert(MetaSchema.fromSchema(schema).toSchema)(hasSameSchemaStructure(schema))
         }
       }
+    ),
+    suite("extended meta schema")(
+      test("represents known type as Known") {
+        val meta1 = ExtendedMetaSchema.fromSchema(Schema[Pet])
+        val meta2 = ExtendedMetaSchema.fromSchema(Schema[DynamicValue])
+        val meta3 = ExtendedMetaSchema.fromSchema(Schema[Recursive])
+
+        assertTrue(
+          meta1.isInstanceOf[ExtensibleMetaSchema.Known[_]],
+          meta2.isInstanceOf[ExtensibleMetaSchema.Known[_]],
+          meta3.isInstanceOf[ExtensibleMetaSchema.Product[_]]
+        )
+      },
+      test("materializes the original schema") {
+        val meta1 = ExtendedMetaSchema.fromSchema(Schema[Pet])
+        val meta2 = ExtendedMetaSchema.fromSchema(Schema[DynamicValue])
+
+        val refEq1 = meta1.toSchema eq Schema[Pet]
+        val refEq2 = meta2.toSchema eq Schema[DynamicValue]
+        assertTrue(refEq1, refEq2)
+      }
```

Comment:
There are no tests for the type list functionality. Could be useful to add them.

Reply:
These two test cases with `ExtendedMetaSchema` are trying to test that - they use `type ExtendedMetaSchema = ExtensibleMetaSchema[DynamicValue :: Pet :: End]` instead of the default. What else would you test?